src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -1002,6 +1002,13 @@ void VM_Version::get_processor_features() {
     }
   }
 
+  if (UseAPX && (UseAVX < 3)) {
```

Comment:
Thanks.  I added a comment.

Reply:
> What about Knight CPUs which have limited AVX512 features?

Any VEX encoded instruction directly accessing an EGPR operand or a memory operand with EGPR BASE / INDEX must be promoted to Extended EVEX encoding. 

Please consider following example:-

CPROMPT>xed64 -64 -d 62 da 7d 28 18 13
62DA7D281813
ICLASS:     VBROADCASTSS
CATEGORY:   BROADCAST
EXTENSION:  AVX512EVEX
IFORM:      VBROADCASTSS_YMMf32_MASKmskw_MEMf32_AVX512
ISA_SET:    AVX512F_256
ATTRIBUTES: DISP8_TUPLE1 MASKOP_EVEX MEMORY_FAULT_SUPPRESSION 
SHORT:      vbroadcastss ymm2, dword ptr [r27]
  
Since broadcast accesses an EGPR register and a non-512 bit vector hence target CPU must also be capable of supporting vector length orthogonality which necessitate AVX512VL feature.

APX in true sense is a mix of both ISA extension (new PUSH/POP2, PPX and NDD instructions, JMPABS, COND COMPARE etc..) and also provisions additional general purpose registers to existing ISA, in the later case user may expect to benefit from costly 3 cycles spills from GPR to XMM enabled with UseFPUForSpilling.